pkg/apis/workflow/v1alpha1/container_set_template_types.go

```diff
@@ -42,6 +46,62 @@ func (in *ContainerSetTemplate) HasSequencedContainers() bool {
 	return false
 }
 
+// Validate checks if the ContainerSetTemplate is valid
+func (in *ContainerSetTemplate) Validate(tmpl *Template) error {
+	if len(in.Containers) == 0 {
+		return fmt.Errorf("containers must have at least one container")
+	}
+
+	names := make([]string, 0)
+	for _, ctr := range in.Containers {
+		names = append(names, ctr.Name)
+	}
+	err := validateWorkflowFieldNames(names, false)
+	if err != nil {
+		return fmt.Errorf("containers%s", err.Error())
+	}
+
+	if len(tmpl.Inputs.Artifacts) > 0 || len(tmpl.Outputs.Parameters) > 0 || len(tmpl.Outputs.Artifacts) > 0 {
```

Comment:
AFAIK it's okay to have `inputs.parameters` without the main container?

For example, this works:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: pod-
  labels:
    workflows.argoproj.io/container-runtime-executor: emissary
spec:
  entrypoint: main
  templates:
  - name: main
    steps:
      - - name: main
          template: print-message
          arguments:
            parameters:
              - name: message
                value: "hello world"

  - name: print-message
    inputs:
      parameters:
        - name: message

    containerSet:
      containers:
        - name: a
          image: argoproj/argosay:v2
          command: ["cowsay"]
          args: ["{{inputs.parameters.message}}"]
```